### ISSUE TITLE
Update docs and storage provisioner  to support updated kubernetes storage class params

### ISSUE THREAD

**Issue by dc232** [NetApp/trident#716]
**Describe the solution you'd like**
Currently according to docs found at 
https://docs.netapp.com/us-en/trident/trident-reference/objects.html#kubernetes-storageclass-objects
The current flag is to incorporate the param
```
    "fsType" = "nfs" 
```
into the storageclass to allow for volumes to be provisioned via sysems such as Azure net App files
sadly this produces a deprecation warning as seen via
https://github.com/NetApp/trident/issues/707
which says
```
"fstype" is deprecated and will be removed in a future release, please use "csi.storage.k8s.io/fstype" instead
```

**Describe alternatives you've considered**
I have tried to add the param into the storage class
```
    "csi.storage.k8s.io/fstype" = "nfs"
````
however this does not seem to be supported by the storage provisioner  and thus it fails to  create the volume for each pod with an attached persistentant volume

**Additional context**
Addiotnal docs that the trident provioner could consume can be found here
https://kubernetes-csi.github.io/docs/external-provisioner.html

**Comment by scaleoutsean** [comment]
> I have tried to add the param into the storage class  "csi.storage.k8s.io/fstype" = "nfs" however this does not seem to be supported by the storage provisioner

That's a claim. But there's no reference, no error log, no configuration files, no version and distro info of Kubernetes, no Trident version and back-end info, and no Storage Class example that substantiates that claim.

` "csi.storage.k8s.io/fstype" = "nfs"` works fine on my Kubernetes v1.23.4 with Trident v22.01.1.

**Comment by jrhunger** [comment]
Maybe this was fixed in the latest version. Below is an example of the failure with 21.04.0 and k8s 1.21.9 (although since the error comes from trident-main i would guess it's the trident version that matters).
```
# sbx-jh1: tridentctl version
t+----------------+----------------+
| SERVER VERSION | CLIENT VERSION |
+----------------+----------------+
| 21.04.0        | 21.04.0        |
+----------------+----------------+
# sbx-jh1: kubectl version
Client Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.5", GitCommit:"aea7bbadd2fc0cd689de94a54e5b7b758869d691", GitTreeState:"clean", BuildDate:"2021-09-15T21:10:45Z", GoVersion:"go1.16.8", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.9", GitCommit:"b631974d68ac5045e076c86a5c66fba6f128dc72", GitTreeState:"clean", BuildDate:"2022-01-19T17:45:53Z", GoVersion:"go1.16.12", Compiler:"gc", Platform:"linux/amd64"}

# sbx-jh1: cat sc-std2-newfstype.yaml
apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
  name: sc-trident-nas-std2
provisioner: csi.trident.netapp.io
parameters:
  backendType: "ontap-nas"
  csi.storage.k8s.io/fstype: "nfs"
  selector: "name=backend-standard"
allowVolumeExpansion: true
reclaimPolicy: Delete

# sbx-jh1: kubectl -n trident logs trident-csi-9bc9c6f68-7zpsn trident-main | tail
time="2022-04-25T15:58:25Z" level=info msg="Found storage class for requested volume pvc-c2ea8f92-ec19-433f-bb29-8e0729464b15." name=sc-trident-nas-std2 requestID=18806ffe-196d-4a6c-a13b-178d85b5a325 requestSource=CSI
time="2022-04-25T15:58:25Z" level=error msg="GRPC error: rpc error: code = Unknown desc = unknown storage class: sc-trident-nas-std2" requestID=18806ffe-196d-4a6c-a13b-178d85b5a325 requestSource=CSI
time="2022-04-25T15:58:30Z" level=info msg="Found PVC for requested volume pvc-c2ea8f92-ec19-433f-bb29-8e0729464b15." UID=c2ea8f92-ec19-433f-bb29-8e0729464b15 name=std2-claim namespace=default requestID=ccc87cb6-8c84-4b23-a86c-d01dc5faf915 requestSource=CSI size="{{10737418240 0} {<nil>} 10Gi BinarySI}" storageClass=sc-trident-nas-std2
time="2022-04-25T15:58:30Z" level=info msg="Found storage class for requested volume pvc-c2ea8f92-ec19-433f-bb29-8e0729464b15." name=sc-trident-nas-std2 requestID=ccc87cb6-8c84-4b23-a86c-d01dc5faf915 requestSource=CSI
time="2022-04-25T15:58:30Z" level=error msg="GRPC error: rpc error: code = Unknown desc = unknown storage class: sc-trident-nas-std2" requestID=ccc87cb6-8c84-4b23-a86c-d01dc5faf915 requestSource=CSI
time="2022-04-25T15:58:33Z" level=warning msg="K8S helper has no record of the updated storage class; instead it will try to create it." name=sc-trident-nas-std2 parameters="map[backendType:ontap-nas csi.storage.k8s.io/fstype:nfs selector:name=backend-standard]" provisioner=csi.trident.netapp.io requestID=62a61ba4-3a3c-445a-a76b-bb1f84c98732 requestSource=Kubernetes
time="2022-04-25T15:58:33Z" level=error msg="K8S helper could not process the storage class attribute csi.storage.k8s.io/fstype" error="unrecognized storage attribute: csi.storage.k8s.io/fstype" name=sc-trident-nas-std2 parameters="map[backendType:ontap-nas csi.storage.k8s.io/fstype:nfs selector:name=backend-standard]" provisioner=csi.trident.netapp.io requestID=62a61ba4-3a3c-445a-a76b-bb1f84c98732 requestSource=Kubernetes

```

**Comment by clintonk** [comment]
Hi, @jrhunger.  This was fixed in this [commit](https://github.com/NetApp/trident/commit/e9901c28358f878c1ce28cd1f24d2a53441624e6) and will be included in Trident 22.04.0.
